### ISSUE TITLE
JWT token accepts none hashing algorithm in the client

### ISSUE THREAD

**Issue by euanmillar** [opencrvs/opencrvs-core#5600]
Its possible to use this approach to create a JWT for a different user and get access to the client to see what the client looks like for different users.  The gateway should still be blocking any data from being manipulated.

JWT library accepts none hashing algorithm. none hashing algorithm is used by the JWT in case the integrity of token is already verified.so an attacker can alter the token claims and token will be trusted by the application.

Fix: Not allowing none hashing algorithm.

https://auth0.com/blog/critical-vulnerabilities-in-json-web-token-libraries/

Solution:

We should use the additional specified algorithm option ...

```

import * as jwt from 'jsonwebtoken'


const options = {
  algorithms: ['RS256'],
  issuer: 'opencrvs:auth-service',
  audience: 'opencrvs:auth-user'
};

function verifyToken(token, secret, options) {
  try {
    const decoded = jwt.verify(token, secret, options);
    return decoded;
  } catch (error) {
    console.error('JWT verification error:', error.message);
    return null;
  }
}
``` 

- [ ] In profileReducer [here](https://github.com/opencrvs/opencrvs-core/blob/ac63063a0e99f44f9e691ee17cc742fc4f28d4f9/packages/client/src/profile/profileReducer.ts#L108). We need to load the publicCert from server docker secret CERT_PUBLIC_KEY_PATH into the client docker container so that the client can access this environment variable and verify the JWT too in the same way that the [gateway does](https://github.com/opencrvs/opencrvs-core/blob/ac63063a0e99f44f9e691ee17cc742fc4f28d4f9/packages/gateway/src/server.ts#L66).  If the JWT cannot be verified, redirectToAuthentication

**Comment by euanmillar** [comment]
For any fix, use Core commit hash 0dce8d7 as a base branch and test in staging before merge, For any fix that requires changes to Farajaland, once merged into develop cherry-pick the fix into [this PR](https://github.com/opencrvs/opencrvs-farajaland/pull/592) for redeployment to the pen test server.

**Comment by naftis** [comment]
I don't think this needs to be fixed right now. Our backend services do check the hashing algorithm, and frontend redirects you to login if the JWT isn't correct.
